File: Python/백준/Silver/1149. RGB거리/RGB거리.py
n = int(input())
rgb = [list(map(int, input().split())) for _ in range(n)]
ans = [[3003]*3 for _ in range(n)]
ans[0] = rgb[0]
for i in range(1,n):
    ans[i][0] = min(ans[i-1][1], ans[i-1][2]) + rgb[i][0]
    ans[i][1] = min(ans[i - 1][0], ans[i - 1][2]) + rgb[i][1]
    ans[i][2] = min(ans[i - 1][0], ans[i - 1][1]) + rgb[i][2]
print(min(ans[n-1]))

File: Java/백준/Silver/1149. RGB거리/RGB거리.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {

	public static void main(String[] args) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int N = Integer.parseInt(br.readLine());
		int[][] RGB = new int[N][3];
		for (int i = 0; i < N; i++) {
			StringTokenizer st = new StringTokenizer(br.readLine()," ");
			for (int j = 0; j < 3; j++) {
				RGB[i][j] = Integer.parseInt(st.nextToken());
			}
		}
		int[][] dp = new int[N+1][3];
		for (int i = 1; i < N+1; i++) {
			dp[i][0] = Math.min(dp[i-1][1], dp[i-1][2]) + RGB[i-1][0];
			dp[i][1] = Math.min(dp[i-1][0], dp[i-1][2]) + RGB[i-1][1];
			dp[i][2] = Math.min(dp[i-1][0], dp[i-1][1]) + RGB[i-1][2];
		}
		System.out.println(Math.min(Math.min(dp[N][0], dp[N][1]), dp[N][2]));
	}

}
